### PR TITLE
track the Global Registry person ID in eDirectory

### PR DESCRIPTION
**Pull request by frett**

### CODE REVIEW

**Comment by frett** [comment]
assigning myself for now until I test this out and make sure things are interacting with it correctly.

File: api/src/main/java/org/ccci/idm/user/dao/ldap/Constants.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
     public static final String LDAP_ATTR_PASSWORDCHANGEDTIME = "pwdChangedTime";
     public static final String LDAP_ATTR_FACEBOOKID = "thekeyFacebookId";
     public static final String LDAP_ATTR_FACEBOOKIDSTRENGTH = "thekeyFacebookIdStrength";
+    public static final String LDAP_ATTR_GRPERSONID = "thekeyGrPersonId";
```
Comment (hlbraddock) [inline api/src/main/java/org/ccci/idm/user/dao/ldap/Constants.java]: I would've thought `grPersonId` and `grStagePersonId` sufficient here.

Comment (frett) [inline api/src/main/java/org/ccci/idm/user/dao/ldap/Constants.java]: this is internal for eDirectory only, I want to keep our custom attributes prefixed with something common for organization.

File: api/src/main/java/org/ccci/idm/user/dao/ldap/AbstractLdapUserDao.java [hunk 2]
```diff
@@ -10,17 +10,19 @@
...
-import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_CRU_PASSWORD_HISTORY;
 import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_CRU_SUB_MINISTRY_CODE;
 import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_DEPARTMENT_NUMBER;
 import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_DOMAINSVISITED;
 import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_EMPLOYEE_NUMBER;
 import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_FACEBOOKID;
 import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_FACEBOOKIDSTRENGTH;
 import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_FIRSTNAME;
+import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_GRPERSONID;
+import static org.ccci.idm.user.dao.ldap.Constants.LDAP_ATTR_GRSTAGEPERSONID;
```
Comment (hlbraddock) [inline api/src/main/java/org/ccci/idm/user/dao/ldap/AbstractLdapUserDao.java]: The Global Registry : a system so foundational that even its staging environment manages to rear its head within the production environment. :)

**Comment by hlbraddock** [comment]
You've added the new attributes to both test and production eDirectory clusters?

**Comment by frett** [comment]
yeah

**Comment by hlbraddock** [comment]
I assumed you probably did, since you've tested, but thought I'd ask (in case not in production yet).

**Comment by hlbraddock** [comment]
+1

**Pull request merged**